src/Build/Evaluation/Expander.cs

```diff
@@ -1512,6 +1516,11 @@ private static object LookupProperty(IPropertyProvider<T> properties, string pro
                 }
                 else
                 {
+                    if (property is ProjectPropertyInstance.EnvironmentDerivedProjectPropertyInstance environmentDerivedProperty)
```

Comment:
do we need to also check for `EnvironmentDerivedProjectProperty` here? or is it OK?

Reply:
Sorry I missed this. I believe it's ok. I looked through uses of EnvironmentDerivedProjectProperties, and I'm fairly confident they're only relevant at evaluation time, at which point our logging context hasn't yet been invalidated, so they get logged then.